deep_speech_2/README.md

```diff
@@ -1,146 +1,393 @@
+It will compute the mean and standard deviation of power spectrum feature with 2000 random sampled audio clips listed in `data/librispeech/manifest.train` and save the results to `data/librispeech/mean_std.npz` for further usage.
+
+
+### Build Vocabulary
+
+A vocabulary of possible characters is required to convert the transcription into a list of token indices for training, and in decoding, to convert from a list of indices back to text again. Such a character-based vocabulary can be built with `tools/build_vocab.py`.
 
 ```
-python tools/compute_mean_std.py --specgram_type mfcc
+python tools/build_vocab.py \
+--count_threshold 0 \
+--vocab_path data/librispeech/eng_vocab.txt \
+--manifest_paths data/librispeech/manifest.train
 ```
 
-and specify ```--specgram_type mfcc``` when running train.py, infer.py, evaluator.py or tune.py.
+It will write a vocabuary file `data/librispeeech/eng_vocab.txt` with all transcription text in `data/librispeech/manifest.train`, without vocabulary truncation (`--count_threshold 0`).
+
+### More Help
 
-More help for arguments:
+For more help on arguments:
 
 ```
+python data/librispeech/librispeech.py --help
 python tools/compute_mean_std.py --help
+python tools/build_vocab.py --help
 ```
 
-### Training
+## Training a model
 
-For GPU Training:
+`train.py` is the main caller of the training module. We show several examples of usage below.
 
-```
-CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 python train.py
-```
+- Start training from scratch with 8 GPUs:
 
-For CPU Training:
+    ```
+    CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 python train.py --trainer_count 8
+    ```
 
-```
-python train.py --use_gpu False
-```
+- Start training from scratch with 16 CPUs:
+
+    ```
+    python train.py --use_gpu False --trainer_count 16
+    ```
+- Resume training from a checkpoint:
+
+    ```
+    CUDA_VISIBLE_DEVICES=0,1,2,3,4,5,6,7 \
+    python train.py \
+    --init_model_path CHECKPOINT_PATH_TO_RESUME_FROM
+    ```
 
-More help for arguments:
+For more help on arguments:
 
 ```
 python train.py --help
 ```
+or refer to `example/librispeech/run_train.sh`.
 
-### Preparing language model
+## Data Augmentation Pipeline
 
-The following steps, inference, parameters tuning and evaluating, will require a language model during decoding.
-A compressed language model is provided and can be accessed by
+Data augmentation has often been a highly effective technique to boost the deep learning performance. We augment our speech data by synthesizing new audios with small random perturbation (label-invariant transformation) added upon raw audios. We don't have to do the syntheses by ourselves, as it is already embedded into the data provider and is done on the fly, randomly for each epoch during training.
 
-```
-cd ./lm
-sh run.sh
-cd ..
-```
+Six optional augmentation components are provided for us to configured and inserted into the processing pipeline.
```

Comment:
Six optional augmentation components are provided which can be configured and inserted into the processing pipeline

Reply:
--> Six optional augmentation components are provided to be selected, configured and inserted into the processing pipeline.